include/otbObiaLSBaatzSegmentationScheduler.txx

```diff
@@ -956,29 +1059,65 @@ LSBaatzSegmentationScheduler<TInputImage>
 
             } // end for(uint32_t tx = 0; tx < nbTilesX; tx++)
 
-            } // end for (uint32_t ty = 0; ty < nbTilesY; ty++)
+        } // end for (uint32_t ty = 0; ty < nbTilesY; ty++)
 
-            // Retrieve the nodes on the borders of the adjacent tiles.
-            auto borderNodeMap = GraphOperationsType::BuildBorderNodesMapForFinalAggregation(this->m_Graph, 
-                                                                                             rowBounds, 
-                                                                                             colBounds, 
-                                                                                             this->m_ImageWidth);
+	
+	// Retrieve the nodes on the borders of the adjacent tiles.
 
+	    std::cout << "MASTER (juste once) : remove duplicate nodes, before : "<< this->m_Graph->GetNumberOfNodes();
+            auto borderNodeMap = GraphOperationsType::BuildBorderNodesMapForFinalAggregation(this->m_Graph,
+                                                                                             rowBounds,
+                                                                                             colBounds,
+                                                                                             this->m_ImageWidth);
             // Remove the duplicated nodes
             GraphOperationsType::RemoveDuplicatedNodes(borderNodeMap, this->m_Graph, this->m_ImageWidth);
 
+	    std::cout << "after : "<< this->m_Graph->GetNumberOfNodes()<<std::endl;
+
             // Update the edges
             GraphOperationsType::DetectNewAdjacentNodes(borderNodeMap, this->m_Graph, this->m_ImageWidth, this->m_ImageHeight);
 
         }
         else
-        {
-            MPI_Send(&m_SerializedStabilityMargin[0],
-                     m_SerializedStabilityMargin.size(),
-                     MPI_CHAR,
-                     0,
-                     0,
-                     MPI_COMM_WORLD);
+	  {
+	    
+	    // BDU : passing by MPI_Type_contiguous for all the proc if at least one proc exceeds the maximum size of an int
+	    if (m_SerializedStabilityMargin.size() > 2147483647)
```

Comment:
If 2147483647 appears several times in the code, store it in a constant and use the constant

Reply:
+1